kernel/src/grant.rs

```diff
@@ -352,34 +257,30 @@ pub struct Iter<'a, T: 'a + Default> {
         core::slice::Iter<'a, Option<&'static dyn ProcessType>>,
         fn(&Option<&'static dyn ProcessType>) -> Option<&'static dyn ProcessType>,
     >,
+    skips_allowed: bool,
```

Comment:
```suggestion
    /// Whether this iterator must visit every grant region, or if
    /// it should skip grant regions which are already entered.
    /// A grant region cannot be entered twice. Therefore, if a capsule
    /// tries to iterate over all grant regions while inside of a grant region,
    /// the kernel will `panic!()` to prevent double entry. However, if this
    /// is set, then the iterator will skip over already entered grant regions,
    /// avoiding a panic but potentially creating subtly unintended code.
    skip_entered_grants: bool,
```